File: backend/src/domain/entities/company.py
from dataclasses import dataclass, field
from datetime import datetime
from decimal import Decimal


@dataclass
class Company:
    name: str
    street: str
    city: str
    state: str
    zip_code: str
    country: str
    billing: Decimal = Decimal("0")
    expenses: Decimal = Decimal("0")
    employees: int = 0
    clients: int = 0
    id: int | None = None
    created_at: datetime = field(default_factory=datetime.utcnow)
    updated_at: datetime = field(default_factory=datetime.utcnow)

    @property
    def profit(self) -> Decimal:
        return self.billing - self.expenses

File: frontend/src/types/company.ts
export type Company = {
    id: number;
    name: string;
    street: string;
    city: string;
    state: string;
    zip_code: string;
    country: string;
    billing: number;
    expenses: number;
    profit: number;
    employees: number;
    clients: number;
    created_at: string;
    updated_at: string;
};

export type CreateCompanyRequest = {
    name: string;
    street: string;
    city: string;
    state: string;
    zip_code: string;
    country: string;
    billing: number;
    expenses: number;
    employees: number;
    clients: number;
};

export type UpdateCompanyRequest = CreateCompanyRequest;

export type PaginatedResponse<T> = {
    items: T[];
    total: number;
    page: number;
    page_size: number;
    total_pages: number;
};

export type CompanyListParams = {
    page?: number;
    page_size?: number;
    search?: string;
    sort_by?: string;
    sort_order?: 'asc' | 'desc';
};
